### DISCUSSION TITLE
Running Flux v2 in read only GIT mode

### DISCUSSION THREAD

**Discussion by Flektoma** [fluxcd/flux2#1819] [Q&A] [answered]
Hi, due to strict restrictions in company environment I am not allowed to use Deploy Key in GitHub for push (just read only). I have no issues with running Flux v2 in readonly mode as I do not expect to use features that would require it.

I have found out that Flux v1 had such option by passing command line arg `--git-readonly` and it was planned to make this feature available in v2 as well https://fluxcd.io/roadmap/#flux-read-only-feature-parity, however there is not working link to docs where it would be explained how to turn it on. 

Is this already supported and if so, how it can be enabled? Thank you

**Comment by stefanprodan** [comment, marked as answer]
When you bootstrap Flux with `flux bootstrap` it generates a read-only deploy key.  We inverted the settings from v1, read-only is the default, to change it you need to specify `flux bootstrap --read-write-key` or provide your ssh key with `--private-key-file`.

**Reply by brokenjacobs** [reply]
Is there a way to make the bootstrap itself read only?

**Reply by stefanprodan** [reply]
Bootstrap implies writing the Flux manifests to upstream so that Flux can upgrade itself. You can use `flux install` which doesn’t touches your repository.

**Reply by brokenjacobs** [reply]
I want flux to be able to upgrade itself, I just want to reproducibly set up flux. I’d like to write the manifests to a repo and have flux use them. Not require a write to a repo on each install. I want the installation to be repeatable using the same persistent repo.

**Reply by stefanprodan** [reply]
You can automate the Flux upgrade from CI without using `bootstrap` every time. Here is an example: https://github.com/fluxcd/flux2/tree/main/action#automate-flux-updates
